src/app/widgets/widget-racer-line/widget-racer-line.component.ts

```diff
@@ -0,0 +1,408 @@
+    this.maxValueTextWidth = Math.floor(this.dToLineElement.width * 0.95);
+    this.maxValueTextHeight = Math.floor(this.dToLineElement.height * 0.95);
+  }
+
+  protected startWidget(): void {
+    this.unsubscribeDataStream();
+    this.dtsValue = null;
+    this.lengthValue = null;
+    this.biasValue = null;
+    this.getColors(this.widgetProperties.config.color);
+    this.observeDataStream('dtsPath', newValue => {
+      this.dtsValue = newValue.data.value;
+      if (this.widgetProperties.config.ignoreZones) {
+        if (!this.dtsValue) {
+          this.dtsColor = this.valueColor;
+        } else if (this.dtsValue < 0) {
+          this.dtsColor = this.theme().zoneAlarm;
+        } else if (this.dtsValue < 10) {
+          this.dtsColor = this.theme().zoneWarn;
+        } else if (this.dtsValue < 20) {
+          this.dtsColor = this.theme().zoneAlert;
+        } else {
+          this.dtsColor = this.valueColor;
+        }
+      } else {
+        switch (newValue.state) {
+          case States.Alarm:
+            this.dtsColor = this.theme().zoneAlarm;
+            break;
+          case States.Warn:
+            this.dtsColor = this.theme().zoneWarn;
+            break;
+          case States.Alert:
+            this.dtsColor = this.theme().zoneAlert;
+            break;
+          default:
+            this.dtsColor = this.valueColor;
+            break;
+        }
+      }
+      this.updateCanvas();
+    });
+
+    this.observeDataStream('lineLengthPath', newValue => {
+      this.lengthValue = newValue.data.value;
+      this.updateCanvas();
+    });
+
+    this.observeDataStream('lineBiasPath', newValue => {
+      this.biasValue = newValue.data.value;
+      this.updateCanvas();
+    });
+
+    this.signalk.subscribeRequest().pipe(takeUntilDestroyed(this.destroyRef)).subscribe(requestResult => {
+      // e.g.
+      // {
+      //   "requestId":"c8ce9f72-5218-4e9b-88fe-4fcc1966574d",
+      //   "state":"COMPLETED",
+      //   "statusCode":200,
+      //   "widgetUUID":"0ad57184-5030-497c-a775-b1cbe06a9d02",
+      //   "message":"Put start line end: OK",
+      //   "statusCodeDescription":"The request was successfully."
+      // },
+      // {
+      //   "requestId":"fc015ba8-1e1a-45dd-a788-75107e334dcd",
+      //   "state":"COMPLETED",
+      //   "statusCode":405,
+      //   "widgetUUID":"0ad57184-5030-497c-a775-b1cbe06a9d02",
+      //   "message":"PUT not supported for navigation.racing.setStartLine",
+      //   "statusCodeDescription":"The server does not support the request."
+      // }
+
+      if (requestResult.widgetUUID === this.widgetProperties.uuid) {
+        console.log('RESULT RECEIVED: ', JSON.stringify(requestResult));
+        if (requestResult.statusCode === 200) {
+          this.beep(600, 50)
+        } else {
+          this.errorMessage = 'Error: ' + requestResult.message;
+          this.mode = -1;
+          this.beep(300, 1000);
+          this.updateCanvas();
+        }
+      }
```

Comment:
private readonly _app = inject(AppService);

Reply:
I now use the snack bar for a general message about the plugin and put the specific error message in the widget